python/foglamp/services/core/syslog_parser.py

```diff
@@ -0,0 +1,91 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+# FOGLAMP_BEGIN
+# See: http://foglamp.readthedocs.io/
+# FOGLAMP_END
+
+from pyparsing import Word, alphas, Combine, nums, string, Regex
+from time import strftime
+
+__author__ = "Praveen Garg"
+__copyright__ = "Copyright (c) 2017 OSIsoft, LLC"
+__license__ = "Apache 2.0"
+__version__ = "${VERSION}"
+
+
+_help = """
+    -------------------------------------------------------------------------------
+    | GET             | /foglamp/syslog                                             |
+    -------------------------------------------------------------------------------
+    """
+
+
+class SyslogParser(object):
+
+    def __init__(self):
+
+        ints = Word(nums)
+
+        # timestamp
+        month = Word(string.ascii_uppercase, string.ascii_lowercase, exact=3)
+        day = ints
+        hour = Combine(ints + ":" + ints + ":" + ints)
+
+        timestamp = month + day + hour
+
+        # hostname
+        hostname = Word(alphas + nums + "_" + "-" + ".")
+
+        # appname
+        appname = Word(" FogLAMP" + alphas + " ")
+
+        # message
+        message = Regex(".*")
+
+        # pattern build
+        self.__pattern = timestamp + hostname + appname + message
+
+    def parse(self, line):
+        # print(line)
+
+        # print(self.__pattern)
+        parsed = self.__pattern.parseString(line)
+
+        payload = dict()
+        payload["timestamp"] = strftime("%Y-%m-%d %H:%M:%S")
+        payload["hostname"] = parsed[3]
+        payload["appname"] = parsed[4]
+        # get [PID] LEVEL: msg from parsed[5]
+        payload["message"] = parsed[5]
+
+        return payload
+
+
+""" --------------------------------- """
+
+
+def logs_main():
+    parser = SyslogParser()
+    logs = []
+
+    with open('/var/log/syslog') as sys_log_file:
+        for line in sys_log_file:
+            fields = parser.parse(line)
+            if "FogLAMP" in fields["appname"]:
+                # print("parsed:", fields)
+                logs.append(fields)
+    return logs
+
+def logs_storage():
+    parser = SyslogParser()
+    logs = []
+
+    with open('/var/log/syslog') as sys_log_file:
+        for line in sys_log_file:
+            fields = parser.parse(line)
+            if "Storage" in fields["appname"]:
+                # print("parsed:", fields)
+                logs.append(fields)
+    return logs
```

Comment:
looks like based on arg above 2 def can be 1.

Remove print from whole file , and make lint happy